provider/CONTRIBUTE.md

```diff
@@ -135,6 +136,119 @@ Once done, you can just press F5 and the debug will start! You can place breakpo
 
 *Note: the first time your start the debug, it can take a while, you need to be patient :-)*
 
+## How to contribute to the Azure RM provider
+
+*Note: we also invite you to read the official documentation about developing a provider [here](https://www.terraform.io/docs/plugins/basics.html#developing-a-plugin)*
+
+First you need to pick [an issue on the provider](https://github.com/terraform-providers/terraform-provider-azurerm/issues) by commenting the issue and saying that you're going to work on that, to make sure that HashiCorp people who maintain the repository and will review your contribution are aware that you are going to work on that.

```

Comment:
update wording to "... to make sure that the repo maintainers are aware that you are going to work on this issue." We should also have them submit the schema that they intend to implement so that feedback can be provided on that before it is implemented. This will save significant rework.